templates/vue/src/components/FileUpload.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <b-container fluid>
+    <b-row>
+        <b-col>
+            <b-form-file
+              ref="file"
+              name="async-upload"
+              class="image-file mb-2"
+              placeholder="Choose a file or drop it here..."
+              drop-placeholder="Drop file here..."
+              @change="uploadFile"
+              required>
+            </b-form-file>
+        </b-col>
+        <b-col sm="1" class="divider">
+          <h6>OR</h6>
+        </b-col>
+        <b-col>
+          <b-row>
+            <b-form-input
+              :id="inputId"
+              :placeholder="placeholderText"
+              :value="value"
+              @input="$emit('input', $event)"
+              required
+            />
+          </b-row>
+        </b-col>
+    </b-row>
+    <b-row>
+      <progress max="100" :value.prop="uploadPercentage"></progress>
+    </b-row>
+  </b-container>
+</template>
+
+<script>
+import axios from "axios"
+
+export default {
+  name: "file-upload",
+  props: {
+    inputId: {
+      type: String,
+      required: true,
+    },
+    placeholderText: {
+      type: String,
+      required: true,
+    },
+    value: {
+      type: String,
+      required: false
+    }
+  },
+
+  data() {
+    return{
+      uploadPercentage: 0
+    }
+  },
+
+  methods: {
+    uploadFile(event) {
+      var formData = new FormData()
```

Comment:
Try to use `const` over `var` whenever possible. `var` is not block scoped so other places may counter-intuitively have access to it.